### ISSUE TITLE
Inconsistency between layer names and weight names in HDF5 saved from tf.keras nested models

### ISSUE THREAD

**Issue by caisq** [tensorflow/tensorflow#19836]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Doesn't matter
- **TensorFlow installed from (source or binary)**: pip
- **TensorFlow version (use command below)**: 1.8.0
- **Python version**:  Doesn't matter

Supposed you run the following code to construct a nested Sequential model with tf.keras and save it as a .h5 file.

```py
import tensorflow as tf


with tf.Graph().as_default(), tf.Session():
  inner_model = tf.keras.Sequential([
      tf.keras.layers.Dense(4, input_shape=[3], activation='relu'),
      tf.keras.layers.Dense(3, activation='tanh')])
  outer_model = tf.keras.Sequential()
  outer_model.add(inner_model)
  outer_model.add(tf.keras.layers.Dense(1, activation='sigmoid'))
  outer_model.compile(loss='binary_crossentropy', optimizer='sgd')

  outer_model.save('/tmp/tf_nested_keras.h5')
```

Inside the saved .h5 file, you can see the layers have the names:
* 'dense_1',
* 'dense_2' 
* 'dense_3'.

However, the weight names do not match up with the layer names:
* 'dense/kernel:0', 'dense/bias:0', 'dense_1/kernel:0', 'dense_1/bias:0', 'dense_2/kernel:0', 'dense_2/bias:0',

This is causing issues for TensorFlow.js converters. Note that the same issue does not occur for non-tf keras.

**Comment by caisq** [comment]
cc @bileschi

**Comment by facaiy** [comment]
I think the issue has been fixed in 693b339a:
> tf.keras.layers is now agnostic to variable scopes and global collections (future-proof). It also uses ResourceVariable everywhere by default.

**Comment by caisq** [comment]
@facaiy Thanks for the info. Yes, I can confirm that this issue is resolved in 1.9.0rc0. Therefore I will close this issue now.
